ethereum-consensus/src/clock.rs

```diff
@@ -196,19 +196,15 @@ impl<T: TimeProvider + Send + Sync> Clock<T> {
 pub type SystemClock = Clock<SystemTimeProvider>;
 
 #[cfg(feature = "async")]
-use tokio_stream::Stream;
+use tokio_stream::{wrappers::IntervalStream, Stream, StreamExt};
 
 #[cfg(feature = "async")]
 impl<T: TimeProvider + Send + Sync> Clock<T> {
     pub fn stream_slots(&self) -> impl Stream<Item = Slot> + '_ {
-        async_stream::stream! {
-            loop {
-                let slot = self.current_slot().expect("after genesis");
-                yield slot;
-                let duration_until_next_slot = self.duration_until_slot(slot + 1);
-                tokio::time::sleep(duration_until_next_slot).await;
-            }
-        }
+        let slot = self.current_slot().expect("after genesis");
+        let duration_until_next_slot = self.duration_until_slot(slot + 1);
+        let interval_stream = IntervalStream::new(tokio::time::interval(duration_until_next_slot));
```

Comment:
this keeps the same amount of offset in the slot for each tick...

I think the best option is to use https://docs.rs/tokio/latest/tokio/time/fn.interval_at.html

So compute the duration until next slot from now, and just have the interval run every `SECONDS_PER_SLOT`, starting from that duration

this changes the API some as we lose a tick for the _current_ slot, but it looks like we would need to implement a custom stream implementation to retain this behavior (would be great if you are up for the challenge :) )

otherwise, lets just make an interval using `interval_at` and add some docs about **not** having a tick for the current slot (which I think is a fine compromise for now...)